File: bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI.Core/SystemRegister/AccessPackage/AccessPackage.cs
using System.ComponentModel.DataAnnotations;

namespace Altinn.Authentication.UI.Core.SystemRegister
{
    /// <summary>
    /// Model describing a complete access package
    /// </summary>
    public class AccessPackage
    {
        /// <summary>
        /// The id of the access package
        /// </summary>
        [Required]
        public required string Id { get; set; }

        /// <summary>
        /// The urn of the access package
        /// </summary>
        [Required]
        public required string Urn { get; set; }

        /// <summary>
        /// The title of service
        /// </summary>
        [Required]
        public string? Name { get; set; }

        public List<ServiceResource> Resources { get; set; } = [];
    }
}

File: frontend/src/types/accessPackage.ts
import { SupportedLanguage } from './language';
import { ServiceResource } from './serviceResource';

export interface AccessPackage {
  id: string;
  urn: string;
  name: SupportedLanguage;
  description: SupportedLanguage;
  area?: {
    id: string;
    name: string;
    description: string;
    iconName: string;
    shortDescription: string;
  };
  resources: ServiceResource[];
}
